![Callysto.ca Banner](https://github.com/callysto/curriculum-notebooks/blob/master/callysto-notebook-banner-top.jpg?raw=true)

<a href="https://hub.callysto.ca/jupyter/hub/user-redirect/git-pull?repo=https%3A%2F%2Fgithub.com%2Fcallysto%2Fhackathon&branch=master&subPath=ColonizingMars/ChallengeTemplates/challenge-option-2-how-could-we-colonize-mars.ipynb&depth=1" target="_parent"><img src="https://raw.githubusercontent.com/callysto/curriculum-notebooks/master/open-in-callysto-button.svg?sanitize=true" width="123" height="24" alt="Open in Callysto"/></a>

# *Sustaining Life on Mars: Data Science Challenge*.


You’re a data scientist on a team of newly-arrived humans. While you were on Earth, you figured out how you could make the planet habitable. From growing food to clothing needs, you need to start building the framework for sustaining life on the red planet. 

In this notebook, we have decided to focus on **Plastics** as key resource that needs to be produced and properly managed.   

![music image](https://github.com/callysto/hackathon/blob/sustainable-society/SustainabilityOnMars/ChallengeExamples/plastic-example-image.jpg?raw=true)

### Section I: Problem background

Plastics are useful for generating fibers (textiles, rope), sheets (bags, wraps, windows), and 3D structures. On Earth, one million plastic drinking bottles are purchased every minute, and up to 5 trillion single-use plastic bags are used worldwide every year. In total, half of all plastic produced is designed to be used only once — and then thrown away (source: UN Environment). 

As a data scientist, we can help researchers understand where the plastic is found and look for innovative ways to recycle plastics and reduce plastic pollution. There is no such thing as a sustainable product in unsustainable packaging.

In this notebook, we want to help scientists reduce  what is not needed so we want to answer the question of **where does tha mjority of plastic waste come from**? we can answer this in many ways, e.g. by sector, or by country.  We will also explore the extent of plastic waste coming from lollipops 🍭🍭🍭. 

### Section II: The data you used

The data in this notebook was downloaded from Citizen Science Cloud: https://hub.cscloud.host/app/ec-2020-plastics. 

It comes from **three** different projects. 

1. **Marine Debris Monitoring and Assessment Project**: a National Oceanic and Atmospheric Administration-coordinated citizen science initiative that engages volunteers to survey and record the amount and types of marine debris on shorelines. 

2. **Ocean Conservancy TIDES Database**: a public, global ocean trash data set, all collected by volunteers. 

3. **European Environment Agency's Marine LitterWatch**: Data collected by the Marine Litter Watch is collected as both clean-ups and as monitoring events. 

Below are some of the fields available for each track which we will use to answer the questions we are interested in:

**CountryName_FromSource** — country name.<br>
**TotalClassifiedItems_EC2020** - equals the sum of columns from [SUM_Hard_PlasticBeverageBottle : SUM_OtherPlasticDebris].<br>
**PCT_PlasticAndFoam** - percentage of plastic and foam from total classified items.<br>
**PCT_Glass_Rubber_Lumber_Metal** - percentage of rubber, lumber and metal from total classified items.<br>
**CONTINENT** - continent.<br>
**SUM_HardPlasticBeverageBottle** - sum of collected plastic beverage bottles.<br>


In [1]:
# importing libraries
import pandas as pd
import matplotlib.pyplot as plt
import cufflinks as cf

In [2]:
# reading the data 
# we have csv file stored in the cloud
url = "https://raw.githubusercontent.com/callysto/hackathon/sustainable-society/SustainabilityOnMars/ChallengeExamples/plastics-data-2015-2018.csv"

# read csv file from url and save it as dataframe
plastics = pd.read_csv(url, low_memory=False)

# print first 5 rows
plastics.head()

UniqueID SourceID                                     LocationFreqID  \
1  MDP-349  40-3153  Blackpoint Beach (Lon -123.4355847 Lat 38.6905...   
2  MDP-351  37-3164  Dune Drift Beach (Lon -123.4844062 Lat 38.7287...   
3  MDP-354  59-3175           Ohlson Beach (Lon -123.4564 Lat 38.7132)   
4  MDP-358  41-3191      Walk On Beach (Lon -123.490915 Lat 38.735105)   
5  MDP-360  11-3195            Rocky Point (Lon -124.4621 Lat 42.7149)   

                          Dataset  \
1  NOAA MDMAP Accumulation Survey   
2  NOAA MDMAP Accumulation Survey   
3  NOAA MDMAP Accumulation Survey   
4  NOAA MDMAP Accumulation Survey   
5  NOAA MDMAP Accumulation Survey   

                                    Organization CountryName_FromSource  \
1  California Coast National Monument Task Force          United States   
2  California Coast National Monument Task Force          United States   
3  California Coast National Monument Task Force          United States   
4  California Coast National Monument Task Force          United States   
5                                  Redfish Rocks          United States   

   Longitude1  Latitude1  Longitude2  Latitude2  ...  \
1 -123.435585  38.690549 -123.432939  38.689234  ...   
2 -123.484406  38.728707 -123.487692  38.733347  ...   
3 -123.456400  38.713200 -123.455100  38.710600  ...   
4 -123.490915  38.735105 -123.489614  38.731897  ...   
5 -124.462100  42.714900 -124.462000  42.713900  ...   

   SUM_Foam_OtherPlasticDebris  SUM_OtherPlasticDebris        NAME  \
1                           13                     1.0  California   
2                            9                     0.0  California   
3                            0                     0.0  California   
4                            0                     0.0  California   
5                          850                     0.0      Oregon   

         COUNTRY  ISO_CODE ISO_CC  ISO_SUB     COUNTRYAFF      CONTINENT  \
1  United States      USCA     US       CA  United States  North America   
2  United States      USCA     US       CA  United States  North America   
3  United States      USCA     US       CA  United States  North America   
4  United States      USCA     US       CA  United States  North America   
5  United States      USOR     US       OR  United States  North America   

      LAND_TYPE  
1  Primary land  
2  Primary land  
3  Primary land  
4  Primary land  
5  Primary land  

[5 rows x 51 columns]

In [3]:
# how many rows and colums does the dataframe have?
plastics.shape

(54388, 51)

### Section III: Data Analysis and Visualization

We want to explore major sources of plastic waste.

Let's look at what plastic items were collected and counted. We do this by checking the column names of the 'plastics' table. 

In [4]:
# print column names
plastics.columns

Index(['UniqueID', 'SourceID', 'LocationFreqID', 'Dataset', 'Organization',
       'CountryName_FromSource', 'Longitude1', 'Latitude1', 'Longitude2',
       'Latitude2', 'TotalWidth_m', 'TotalLength_m', 'ShorelineName',
       'EventType', 'TotalVolunteers', 'MonthYear', 'Year', 'MonthNum',
       'Month', 'TotalClassifiedItems_EC2020', 'PCT_PlasticAndFoam',
       'PCT_Glass_Rubber_Lumber_Metal', 'SUM_Hard_PlasticBeverageBottle',
       'SUM_Hard_OtherPlasticBottle', 'SUM_HardOrSoft_PlasticBottleCap',
       'SUM_PlasticOrFoamFoodContainer', 'SUM_Hard_BucketOrCrate',
       'SUM_Hard_Lighter', 'SUM_OtherHardPlastic',
       'SUM_PlasticOrFoamPlatesBowlsCup', 'SUM_HardSoft_PersonalCareProduc',
       'SUM_HardSoftLollipopStick_EarBu', 'SUM_Soft_Bag',
       'SUM_Soft_WrapperOrLabel', 'SUM_Soft_Straw', 'SUM_Soft_OtherPlastic',
       'SUM_Soft_CigaretteButts', 'SUM_Soft_StringRingRibbon', 'Fishing_Net',
       'SUM_FishingLineLureRope', 'Fishing_BuoysAndFloats',
       'SUM_Foam_OtherPl

Columns starting with the prefix '**SUM_**' indicates the type of plastic item that was collected and then counted. These are: e.g. beverage bottles, bottle caps, food containers, buckets, crates, lighters, bowls, cups, personal care products, wappers, labels, straws, cigarette butts, string ring ribbons, fishing nets, fishing line lure ropes, fishing buoys and floats. 

The column '**TotalClassifiedItems_EC2020**' contains the total number of plastic waste sources that have been collected and then classified. 

Let's first get the highest number of plastic items that were collected. We do this by finding the largest number in the column 'TotalClassifiedItems_EC2020' using the max() function. 

In [5]:
# find largest number of plastic items collected and classified
plastics['TotalClassifiedItems_EC2020'].max()

26420613

The highest number of plastic items collected was 26420613. That's a lot!

Next, let's find out more about where this plastic was collected from.

In [6]:
# First let's sort the data by the 'TotalClassifiedItems_EC2020' column by descending order i.e. highest is first
plastics.sort_values(by="TotalClassifiedItems_EC2020", ascending = False)

UniqueID SourceID                                     LocationFreqID  \
50185  TID-51979    66013  TIDES (Lon -0.18885935584342 Lat 5.6160064497862)   
50191  TID-51985    66019   TIDES (Lon 0.89120190773031 Lat 5.7817777402577)   
50183  TID-51977    66011  TIDES (Lon -0.22499734414168 Lat 5.5290572931819)   
50182  TID-51976    66010  TIDES (Lon -0.098134654142427 Lat 5.5794431263...   
50186  TID-51980    66014    TIDES (Lon 0.7696046624684 Lat 5.7716927421673)   
...          ...      ...                                                ...   
39963  TID-40801    49063                  TIDES (Lon 103.08908 Lat 5.40554)   
24774  TID-23205    28544                 TIDES (Lon -80.05129 Lat 26.50476)   
42090  TID-43375    51677   TIDES (Lon -80.065894302447 Lat 26.622105689591)   
19269  TID-16884    21969                   TIDES (Lon 1.57788 Lat 50.51594)   
38756  TID-39295    44929                TIDES (Lon -156.44679 Lat 20.75225)   

                              Dataset                     Organization  \
50185  Oecan Conservancy TIDES Report      SNFYVF- Accra, Volta Region   
50191  Oecan Conservancy TIDES Report                   SNFYVF- Anloga   
50183  Oecan Conservancy TIDES Report  SNFYVF- Accra, Korle Gono Beach   
50182  Oecan Conservancy TIDES Report      SNFYVF- Accra, Teshie Beach   
50186  Oecan Conservancy TIDES Report             SNFYVF- Accra, Dzita   
...                               ...                              ...   
39963  Oecan Conservancy TIDES Report                 2 Marine Biology   
24774  Oecan Conservancy TIDES Report       Sea turtle advfe ntupre se   
42090  Oecan Conservancy TIDES Report              Highland Elementary   
19269  Oecan Conservancy TIDES Report                              NaN   
38756  Oecan Conservancy TIDES Report                        Maui now    

      CountryName_FromSource  Longitude1  Latitude1  Longitude2  Latitude2  \
50185                  Ghana    -0.18886    5.61601         NaN        NaN   
50191                  Ghana     0.89120    5.78178         NaN        NaN   
50183                  Ghana    -0.22500    5.52906         NaN        NaN   
50182                  Ghana    -0.09813    5.57944         NaN        NaN   
50186                  Ghana     0.76960    5.77169         NaN        NaN   
...                      ...         ...        ...         ...        ...   
39963               Malaysia   103.08908    5.40554         NaN        NaN   
24774                    USA   -80.05129   26.50476         NaN        NaN   
42090                    USA   -80.06589   26.62211         NaN        NaN   
19269                 France     1.57788   50.51594         NaN        NaN   
38756                    USA  -156.44679   20.75225         NaN        NaN   

       ...  SUM_Foam_OtherPlasticDebris  SUM_OtherPlasticDebris  \
50185  ...                      8456479              15234016.0   
50191  ...                      4112429               6114136.0   
50183  ...                      2354117               4115426.0   
50182  ...                      2119142               4471246.0   
50186  ...                      1324121               2174997.0   
...    ...                          ...                     ...   
39963  ...                            0                     0.0   
24774  ...                            0                     0.0   
42090  ...                            0                     0.0   
19269  ...                            0                     0.0   
38756  ...                            0                     0.0   

                     NAME        COUNTRY  ISO_CODE ISO_CC  ISO_SUB  \
50185       Greater Accra          Ghana      GHAA     GH       AA   
50191               Volta          Ghana      GHTV     GH       TV   
50183       Greater Accra          Ghana      GHAA     GH       AA   
50182       Greater Accra          Ghana      GHAA     GH       AA   
50186               Volta          Ghana      GHTV     GH       TV

Note the row number associated with the highest items which is 50185

In [7]:
# Now let's have a closer look at the data associated with the highest plastic items, row number 50185
plastics.loc[50185]

UniqueID                                                                   TID-51979
SourceID                                                                       66013
LocationFreqID                     TIDES (Lon -0.18885935584342 Lat 5.6160064497862)
Dataset                                               Oecan Conservancy TIDES Report
Organization                                             SNFYVF- Accra, Volta Region
CountryName_FromSource                                                         Ghana
Longitude1                                                                  -0.18886
Latitude1                                                                    5.61601
Longitude2                                                                       NaN
Latitude2                                                                        NaN
TotalWidth_m                                                                     NaN
TotalLength_m                                                    

#### Which country had the maximum number of plastic waste items collected and classified?

#### Observations
We can see here that the highest number of plastic waste items collected comes from Ghana in Africa. 

Does this mean that Ghana has the highest plastic pollution? Think about it then check the conclusion section for some insights. 

Let's breakdown this number and look at the contribution of each collected plastic item that led to Ghana having the highest number of collected plastic items. We will draw a barplot to visualise the collected plastic items and their counts.

In [ ]:
# let's first select the columns and rows we are interested in
# select the columns containing the required data
maximum_plastics_waste = pd.DataFrame(plastics.loc[50185, 'SUM_Hard_PlasticBeverageBottle':'SUM_OtherPlasticDebris'])

# rename column name to reflect the data within
maximum_plastics_waste.columns = ['plastic_waste_sources']

We will create barplot to visualise the number of each collected plastic item in Ghana. 

In [ ]:
# maximum_plastics_waste.plot(kind = 'bar')
cf.go_offline()
maximum_plastics_waste.iplot(kind='bar', title='What are the major sources of plastic waste in Ghana?', yTitle='type of plastic item', xTitle='number of plastic items collected')

#### Observations
We can see here that the biggest sources of plastic pollution in Ghana, comes from foam plastic debris and foam.  Cigarette butts, and soft straw also contribute plastic waste sources in Ghana. 

We also observe that the columns 'SUM_OtherPlasticDebris' and 'SUM_Foam_OtherPlasticDebris' are significanlty larger than the rest of the values and are very general that we can not tell which type of plastic items is this. 

What happens when we recreate the bar plot without these two columns.  

Let's redraw the barplot witout the 'Sum_OtherPlasticDebris' and 'SUM_Foam_OtherPlasticDebris' columns. 

In [ ]:
# first we drop the columns we are not interested in
maximum_plastics_waste = maximum_plastics_waste.drop(['SUM_OtherPlasticDebris', 'SUM_Foam_OtherPlasticDebris'], axis = 0)

In [ ]:
# then we redraw the bar plot
maximum_plastics_waste.iplot(kind='bar', title='What are the major sources of plastic waste in Ghana?', yTitle='type of plastic item', xTitle='number of plastic items collected')

#### Observations
We observe that soft cigarette butts, soft straws, plates, bowls and cups are the major contributors to plastic waste in Ghana, Africa.

### What about lollipop sticks? Do we have to give up lollipops on Mars?

Notice that lollipop sticks and ear buds are counted and classified among the plastic items that contribute to plastic waste. 

We would like to answer the question of how much do lollipop sticks actually contribute to plastic waste. 

To answer this question, let's look at the distribution of lollipop stick waste across continents. 

We will draw a scatter plot for the count of lollipo sticks in every continent using the columns 'SUM_HardSoftLollipopStick_EarBu' and 'CONTINENT'

In [ ]:
# draw the scatter plot
columns = ["SUM_HardSoftLollipopStick_EarBu"]
for col in columns:
    x = plastics.groupby("CONTINENT")[col].mean()
    
x.iplot(kind='line', title='How much do lollipop sticks contribute to plastic waste?', yTitle='continent', xTitle='number of lollipop sticks and ear buds')

#### Observations
We observe that Europeans have lollipop sticks (and ear buds) as contributing to plastic waste much more than other continents

Does this make sense? Does this mean people in all continents but Europe, don't eat lollipops?

Of course NOT! People all over the globe eat lollipops.

Rememeber that this was an aggregated dataset from three different sources, so let's look at counts of lollipops from each of the three datasets. 

In [ ]:
# plot counts of lollipops by data source
columns = ["SUM_HardSoftLollipopStick_EarBu"]
for col in columns:
    x = plastics.groupby("Dataset")[col].mean()

x.iplot(kind='bar', title='How many lollipop sticks and ear buds were collected in each dataset? ', yTitle='lollipop sticks and ear buds counts', xTitle='dataset')

### Section IV: Conclusion 

We have used data to know that Ghana has the highest number of collected plastic waste items. While this can imply that Ghana has the highest plastic waste across al countries, it could be a limitation of this particular dataset where the number of volunteers could be higher or there could be more plastic waste on particular dates (e.g. holidays). 

We have also seen that lollipos sticks and ear buds were only collected and counted by the 'European Environment Agency's Marine LitterWatch' project which falsely implied that lollipop sticks are source of plastic waste only in Euorpe. 

### What's next ?

Several other questions can be answered with this dataset like how does the trend in sources of plastic waste vary across the years or months or regions? 

📌 In this **example notebook** you have seen how data can give us insights into questions and solutions that are both data-driven. Now go to the **hackathon template** and start solving your own challenge for sustaining life on Mars!